build-npm/build.sh

```diff
@@ -0,0 +1,346 @@
+run_standard_pipeline() {
+  local enable_sonar="${1:-true}"
+  local enable_deploy="${2:-true}"
+  shift 2  # Remove first two parameters
+  local sonar_args=("$@")  # Remaining parameters are sonar args
+
+  run_npm_install
+  run_npm_tests
+
+  if [ "${enable_sonar}" = "true" ]; then
+    run_sonar_scanner "${sonar_args[@]}"
+  fi
+
+  run_npm_build
+
+  if [ "${enable_deploy}" = "true" ]; then
+    jfrog_npm_publish
+  fi
+}
+
+build_npm() {
+  echo "=== NPM Build, Deploy, and Analyze ==="
+  echo "Branch: ${GITHUB_REF_NAME}"
+  echo "Pull Request: ${PULL_REQUEST}"
+  echo "Deploy Pull Request: ${DEPLOY_PULL_REQUEST}"
+  echo "Skip Tests: ${SKIP_TESTS}"
+
+  if is_main_branch && ! is_pull_request; then
+    echo "======= Building main branch ======="
+
+    local current_version
+    current_version=$(get_current_version)
+    echo "Current version: ${current_version}"
+
+    check_version_format "${PROJECT_VERSION}"
+    echo "Checked version format: ${PROJECT_VERSION}."
+
+    run_standard_pipeline true true -Dsonar.projectVersion="${current_version}"
+
+  elif is_maintenance_branch && ! is_pull_request; then
+    echo "======= Building maintenance branch ======="
+
+    local current_version
+    current_version=$(get_current_version)
+
+    if [[ ${current_version} =~ "-SNAPSHOT" ]]; then
+      echo "======= Found SNAPSHOT version ======="
+      set_npm_version_with_build_id "${BUILD_NUMBER}"
+      echo "Set npm version with build ID: ${BUILD_NUMBER}."
+      check_version_format "${PROJECT_VERSION}"
+    else
+      echo "======= Found RELEASE version ======="
+      echo "======= Deploy ${current_version} ======="
+      check_version_format "${current_version}"
+      PROJECT_VERSION="${current_version}"
+      echo "project-version=${PROJECT_VERSION}" >> "${GITHUB_OUTPUT}"
+    fi
+
+    run_standard_pipeline true true -Dsonar.branch.name="${GITHUB_REF_NAME}"
+
+  elif is_pull_request; then
+    echo "======= Building pull request ======="
+
+    if [ "${DEPLOY_PULL_REQUEST:-false}" == "true" ]; then
+      echo "======= with deploy ======="
+      check_version_format "${PROJECT_VERSION}"
+      run_standard_pipeline true true -Dsonar.analysis.prNumber="${PULL_REQUEST}"
+    else
+      echo "======= no deploy ======="
+      run_standard_pipeline true false -Dsonar.analysis.prNumber="${PULL_REQUEST}"
+    fi
+
+  elif is_dogfood_branch && ! is_pull_request; then
+    echo "======= Build dogfood branch ======="
+
+    check_version_format "${PROJECT_VERSION}"
+    run_standard_pipeline false true
+
+  elif is_long_lived_feature_branch && ! is_pull_request; then
+    echo "======= Build long-lived feature branch ======="
+    run_standard_pipeline true false -Dsonar.branch.name="${GITHUB_REF_NAME}"
+
+  else
+    echo "======= Build other branch ======="
+    run_standard_pipeline false false
+  fi
+
+  echo "=== Build completed successfully ==="
+}
+
+main() {
```

Comment:
Can you rename it to build_npm and merge build_npm into it?
After a few tries in the other actions, there's no need for a `build_*` sub-function here.

That revealed to be potentially useful and avoids the generic main function name.